### PR TITLE
[ISSUE 545] Enable Storybook Github Action

### LINKED ISSUE #545: [Task]: Enable Storybook Github Action
### Summary

We have a GA to deploy Storybook to Github Pages. Let's enable it and then share out a link.

### Acceptance criteria

- Enable GA for Storybook Deploy
- Share link

### PR DESCRIPTION
**Pull request by daphnegold**
## Summary
Fixes #545

## Changes proposed
- Add `assetPath` utility
- Update all absolute img paths that aren't using es6 import to use `assetPath` util
- Enable Github Action to deploy Storybook
- Update Github settings for Pages

We have some inconsistencies with img paths. Suggesting a convention where we primarily use Option 1 below, but there seems to be a use case for Option 2. Option 3 is the no-go.

1. Adding `NEXT_PUBLIC_BASE_PATH` as a prefix to the img path
This works, and we have precedence to use `NEXT_PUBLIC_BASE_PATH` as prefix and that is how it is done in the template repo

2. es6 import which is then picked up by webpack. Useful for `<Image />` tags that seem to require a width but not required if imported?
While using es6 imports works because of integration with webpack, and we have one instance of that in our repo, it can also be complicated as a solution if we have to import many images from an array or something (see pdfs). 

3. Just absolute img path without import or `NEXT_PUBLIC_BASE_PATH` as prefix
This can lead to broken img tags, which we see in Storybook where we have a set value for base path. The reason for this is that the actual image resides at https://hhs.github.io/grants-equity/img/grants-gov-logo.png but the it's looking for it at https://hhs.github.io/img/grants-gov-logo.png since the base path is missing.

Even though we are only seeing issues with Storybook (we don't set `NEXT_PUBLIC_BASE_PATH` for our app currently), probably best to be consistent with this to avoid issues in the future.

## Context for reviewers
To test, make sure app looks good with `assetPath` update locally. Here's a link to the Github Page: https://hhs.github.io/grants-equity.

### Updated Github settings here to enable deploy from GA
Screenshot 2023-09-27 at 2 20 59 PM

### Broken Link 
Screenshot 2023-09-27 at 5 57 02 PM

### Not Broken Link
Screenshot 2023-09-27 at 6 10 10 PM

### CODE REVIEW

File: frontend/src/components/Header.tsx [hunk 1]
```diff
@@ -52,9 +54,7 @@ const Header = ({ logoPath, primaryLinks }: Props) => {
                   <span className="margin-right-1">
                     <img
                       className="width-3 desktop:width-5 text-bottom margin-right-05"
-                      src={`${
```
Comment (daphnegold) [inline frontend/src/components/Header.tsx:55]: Hello, precedence

File: frontend/src/pages/_app.tsx [hunk 2]
```diff
@@ -9,15 +9,13 @@ import Layout from "../components/Layout";
...
 
+import { assetPath } from "src/utils/assetPath";
+
 function MyApp({ Component, pageProps }: AppProps) {
   return (
     <>
       <Head>
-        <link
-          rel="icon"
-          href={`${process.env.NEXT_PUBLIC_BASE_PATH ?? ""}/img/favicon.ico`}
```
Comment (daphnegold) [inline frontend/src/pages/_app.tsx:18]: More precedence :P

Comment (daphnegold) [inline frontend/src/pages/_app.tsx:18]: More precedence :P

**Comment by andycochran** [comment]
Does `process.env.NEXT_PUBLIC_BASE_PATH ?? ""` not work for Storybook image paths in the [platform-test-nextjs](https://github.com/navapbc/platform-test-nextjs/blob/866c3b832b123b42da37a3a37d15ff7bc1905504/app/src/components/Header.tsx#L52-L58) template? Do we need to set an environment variable somewhere?

**Comment by daphnegold** [comment]
> Does `process.env.NEXT_PUBLIC_BASE_PATH ?? ""` not work for Storybook image paths in the [platform-test-nextjs](https://github.com/navapbc/platform-test-nextjs/blob/866c3b832b123b42da37a3a37d15ff7bc1905504/app/src/components/Header.tsx#L52-L58) template? Do we need to set an environment variable somewhere?

I'm not sure I understand the question entirely, but it does work. That is one of the images that was functioning, as it used the base path as a prefix. I pulled that logic out into the `assetPath` function so we don't need to repeat it over and over and can use it consistently, then replaced it in the `Header` component. An example of an image that was broken -- pdfs and footer absolute paths.

**Comment by daphnegold** [comment]
> I assume the issue is that gh-pages deploys to a subdirectory?

Yes, and then we set that with `NEXT_PUBLIC_BASE_PATH=${{ steps.pages_config.outputs.base_path }} npm run storybook-build`

File: .github/workflows/cd-storybook.yml [hunk 3]
```diff
@@ -2,11 +2,10 @@ name: Deploy Storybook
...
   # Runs on pushes targeting the default branch
-  # !! Uncomment the lines below to enable this workflow
-  # push:
-  #   branches: ["main"]
-  #   paths:
-  #     - frontend/**
+  push:
+    branches: ["main"]
+    paths:
+      - frontend/**
```
Comment (SammySteiner) [inline .github/workflows/cd-storybook.yml:8]:
```suggestion
      - frontend/.storybook/**
      - frontend/src/**
      - frontend/stories/**
      - .github/workflows/cd-storybook.yml
```
Maybe we don't need to rebuild storybook for all frontend repo changes just storybook relevant ones?

File: frontend/src/utils/assetPath.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+export function assetPath(relativePath: string) {
```
Comment (SammySteiner) [inline frontend/src/utils/assetPath.ts:1]: Thanks for pulling this out into a util!

**Review by SammySteiner — APPROVED** [review]
LGTM! 
One nit with a suggestion.
Thanks for getting storybook sorted out!

File: .github/workflows/cd-storybook.yml (hunk 3 above)
Comment (daphnegold) [reply .github/workflows/cd-storybook.yml:8]: I'm okay leaving it as `frontend/**`. Almost every folder has some relation to Storybook. `/public` houses images and locale strings, so we'd want that too. We likely want to integrate the Storybook test runner add-on. There's not much else at that point that warrants getting this granular. Going to skip committing this, but open to adding it.

**Pull request merged**